### DISCUSSION TITLE
[4.0.0] Require users to explicitly define StringType, etc.

### DISCUSSION THREAD

**Discussion by falbrechtskirchinger** [nlohmann/json#3603] [Ideas]
I'm in the process of debugging #3602. It seems that it would be beneficial if users were required to explicitly tell the library what types are StringTypes, etc.

Just because a type is constructible from a string doesn't make it a string type. This can lead to the compiler selecting (or trying to select and failing) the wrong `from_json` function.

Leaving this here to be fleshed out at a later date.

**Comment by nlohmann** [comment]
Looking into it.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
